pkg/scalers/honeycomb_scaler.go

```diff
@@ -0,0 +1,300 @@
+	v2 "k8s.io/api/autoscaling/v2"
+	"k8s.io/metrics/pkg/apis/external_metrics"
+
+	"github.com/kedacore/keda/v2/pkg/scalers/scalersconfig"
+	kedautil "github.com/kedacore/keda/v2/pkg/util"
+)
+
+const (
+	honeycombScalerName        = "honeycomb"
+	honeycombBaseURL           = "https://api.honeycomb.io/1"
+	maxPollAttempts            = 5
+	initialPollDelay           = 2 * time.Second
+	honeycombQueryResultsLimit = 10000
+)
+
+type honeycombScaler struct {
+	metricType v2.MetricTargetType
+	metadata   honeycombMetadata
+	httpClient *http.Client
+	logger     logr.Logger
+}
+
+type honeycombMetadata struct {
+	APIKey              string                 `keda:"name=apiKey, order=authParams;triggerMetadata"`
+	Dataset             string                 `keda:"name=dataset, order=triggerMetadata"`
+	Query               map[string]interface{} `keda:"name=query, order=triggerMetadata, optional"`
+	QueryRaw            string                 `keda:"name=queryRaw, order=triggerMetadata, optional"`
+	ResultField         string                 `keda:"name=resultField, order=triggerMetadata, optional"`
+	ActivationThreshold float64                `keda:"name=activationThreshold, order=triggerMetadata, default=0"`
+	Threshold           float64                `keda:"name=threshold, order=triggerMetadata"`
+	Breakdowns          []string               `keda:"name=breakdowns, order=triggerMetadata, optional"`
+	Calculation         string                 `keda:"name=calculation, order=triggerMetadata, default=COUNT"`
+	Limit               int                    `keda:"name=limit, order=triggerMetadata, default=1"`
+	TimeRange           int                    `keda:"name=timeRange, order=triggerMetadata, default=60"`
+	TriggerIndex        int
+}
+
+func NewHoneycombScaler(config *scalersconfig.ScalerConfig) (Scaler, error) {
+	metricType, err := GetMetricTargetType(config)
+	if err != nil {
+		return nil, fmt.Errorf("error getting scaler metric type: %w", err)
+	}
+
+	logger := InitializeLogger(config, fmt.Sprintf("%s_scaler", honeycombScalerName))
+
+	meta, err := parseHoneycombMetadata(config)
+	if err != nil {
+		return nil, fmt.Errorf("error parsing honeycomb metadata: %w", err)
+	}
+
+	logger.Info("Initializing Honeycomb Scaler", "dataset", meta.Dataset)
+
+	return &honeycombScaler{
+		metricType: metricType,
+		metadata:   meta,
+		// Query Results cannot take longer than 10 seconds to run, see https://api-docs.honeycomb.io/api for details
+		httpClient: &http.Client{Timeout: 15 * time.Second},
+		logger:     logger,
+	}, nil
+}
+
+func parseHoneycombMetadata(config *scalersconfig.ScalerConfig) (honeycombMetadata, error) {
+	meta := honeycombMetadata{}
+	err := config.TypedConfig(&meta)
+	if err != nil {
+		return meta, fmt.Errorf("error parsing honeycomb metadata: %w", err)
+	}
+	meta.TriggerIndex = config.TriggerIndex
+
+	// Use queryRaw if provided, else build query from legacy fields
+	if raw, ok := config.TriggerMetadata["queryRaw"]; ok && raw != "" {
```

Comment:
Can't we just use the parsed value?
```suggestion
	if meta.QueryRaw != "" {
```